sdcm/nemesis.py

```diff
@@ -1716,9 +1713,42 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
         finally:
             self.target_node.remoter.run("sudo /sbin/ifup eth1")
 
+    def disrupt_grow_shrink_cluster(self):
+        timeout_before_shrink = self.tester.params.get("nemesis_interval")
+        init_nodes_number = self.tester.params.get("n_db_nodes")
+        add_nodes_number = self.tester.params.get('add_node_cnt')
+
+        max_nodes_number = init_nodes_number + add_nodes_number
+
+        added_nodes = []
+        initial_target_node = self.target_node
+
+        self._set_current_disruption("GrowCluster")
+        self.log.info("Start grow cluster from %s to %s", init_nodes_number, max_nodes_number)
+        while len(self.cluster.nodes) < max_nodes_number:
+            self.metrics_srv.event_start('add_node')
+            added_node = self._add_and_init_new_cluster_node()
+            self.metrics_srv.event_stop('add_node')
+            self.log.info("New node added %s", added_node.name)
+            added_nodes.append(added_node)
+        self.log.info("Finish cluster grow")
+
+        time.sleep(timeout_before_shrink)
+
+        self._set_current_disruption("ShrinkCLuster")
+        self.log.info("Start shrink cluster from  %s to %s", max_nodes_number, init_nodes_number)
+        for remove_node in added_nodes:
```

Comment:
You don't need remove specifically the.nodes that you added,  you can remove any node from the cluster.

Reply:
fixed